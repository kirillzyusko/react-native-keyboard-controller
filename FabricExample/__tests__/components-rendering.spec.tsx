import { render } from "@testing-library/react-native";
import React from "react";
import { View } from "react-native";
import {
  KeyboardAvoidingView,
  KeyboardAwareScrollView,
  KeyboardControllerView,
  KeyboardProvider,
  KeyboardStickyView,
  KeyboardToolbar,
} from "react-native-keyboard-controller";

function EmptyView() {
  return <View style={{ width: 20, height: 20, backgroundColor: "black" }} />;
}

function KeyboardControllerViewTest() {
  return <KeyboardControllerView statusBarTranslucent />;
}

function KeyboardProviderTest() {
  return (
    <KeyboardProvider statusBarTranslucent>
      <EmptyView />
    </KeyboardProvider>
  );
}

const style = { marginBottom: 20 };

function KeyboardAvoidingViewTest() {
  return (
    <KeyboardAvoidingView behavior="height" style={style}>
      <EmptyView />
    </KeyboardAvoidingView>
  );
}

function KeyboardAwareScrollViewTest() {
  return (
    <KeyboardAwareScrollView
      bottomOffset={20}
      enabled={true}
      disableScrollOnKeyboardHide={false}
      style={style}
    >
      <EmptyView />
    </KeyboardAwareScrollView>
  );
}

const offset = { closed: -20, opened: -40 };

function KeyboardStickyViewTest() {
  return (
    <KeyboardStickyView offset={offset} style={style}>
      <EmptyView />
    </KeyboardStickyView>
  );
}

function KeyboardToolbarTest() {
  return <KeyboardToolbar content={<EmptyView />} />;
}

describe("components rendering", () => {
  it("should render `KeyboardControllerView`", () => {
    expect(render(<KeyboardControllerViewTest />)).toMatchSnapshot();
  });

  it("should render `KeyboardProvider`", () => {
    expect(render(<KeyboardProviderTest />)).toMatchSnapshot();
  });

  it("should render `KeyboardAvoidingView`", () => {
    expect(render(<KeyboardAvoidingViewTest />)).toMatchSnapshot();
  });

  it("should render `KeyboardAwareScrollView`", () => {
    expect(render(<KeyboardAwareScrollViewTest />)).toMatchSnapshot();
  });

  it("should render `KeyboardStickyView`", () => {
    expect(render(<KeyboardStickyViewTest />)).toMatchSnapshot();
  });

  it("should render `KeyboardToolbar`", () => {
    expect(render(<KeyboardToolbarTest />)).toMatchSnapshot();
  });
});
