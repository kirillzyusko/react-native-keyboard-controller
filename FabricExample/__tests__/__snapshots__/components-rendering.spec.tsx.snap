// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components rendering should render \`KeyboardAvoidingView\` 1`] = `
<View
  behavior="height"
  style={
    {
      "marginBottom": 20,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "black",
        "height": 20,
        "width": 20,
      }
    }
  />
</View>
`;

exports[`components rendering should render \`KeyboardAwareScrollView\` 1`] = `
<RCTScrollView
  bottomOffset={20}
  disableScrollOnKeyboardHide={false}
  enabled={true}
  style={
    {
      "marginBottom": 20,
    }
  }
>
  <View>
    <View
      style={
        {
          "backgroundColor": "black",
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
</RCTScrollView>
`;

exports[`components rendering should render \`KeyboardControllerView\` 1`] = `
<KeyboardControllerView
  statusBarTranslucent={true}
/>
`;

exports[`components rendering should render \`KeyboardProvider\` 1`] = `
<KeyboardProvider
  statusBarTranslucent={true}
>
  <View
    style={
      {
        "backgroundColor": "black",
        "height": 20,
        "width": 20,
      }
    }
  />
</KeyboardProvider>
`;

exports[`components rendering should render \`KeyboardStickyView\` 1`] = `
<View
  offset={
    {
      "closed": -20,
      "opened": -40,
    }
  }
  style={
    {
      "marginBottom": 20,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "black",
        "height": 20,
        "width": 20,
      }
    }
  />
</View>
`;

exports[`components rendering should render \`KeyboardToolbar\` 1`] = `
<View
  content={
    {
      "$$typeof": Symbol(react.transitional.element),
      "_owner": FiberNode {
        "_debugHookTypes": null,
        "_debugInfo": null,
        "_debugNeedsRemount": false,
        "_debugOwner": null,
        "actualDuration": -0,
        "actualStartTime": 1740238588745,
        "alternate": null,
        "child": FiberNode {
          "_debugHookTypes": null,
          "_debugInfo": null,
          "_debugNeedsRemount": false,
          "_debugOwner": [Circular],
          "actualDuration": -0,
          "actualStartTime": 1740238588745,
          "alternate": null,
          "child": FiberNode {
            "_debugHookTypes": null,
            "_debugInfo": null,
            "_debugNeedsRemount": false,
            "_debugOwner": [Circular],
            "actualDuration": -0,
            "actualStartTime": 1740238588726,
            "alternate": null,
            "child": null,
            "childLanes": 0,
            "deletions": null,
            "dependencies": null,
            "elementType": "View",
            "flags": 0,
            "index": 0,
            "key": null,
            "lanes": 0,
            "memoizedProps": {
              "children": undefined,
              "content": [Circular],
            },
            "memoizedState": null,
            "mode": 1,
            "pendingProps": {
              "children": undefined,
              "content": [Circular],
            },
            "ref": null,
            "refCleanup": null,
            "return": [Circular],
            "selfBaseDuration": -0,
            "sibling": null,
            "stateNode": {
              "children": [],
              "internalInstanceHandle": [Circular],
              "isHidden": false,
              "props": {
                "children": undefined,
                "content": [Circular],
              },
              "rootContainerInstance": {
                "children": [
                  [Circular],
                ],
                "createNodeMock": [Function],
                "tag": "CONTAINER",
              },
              "tag": "INSTANCE",
              "type": "View",
            },
            "subtreeFlags": 0,
            "tag": 5,
            "treeBaseDuration": -0,
            "type": "View",
            "updateQueue": null,
          },
          "childLanes": 0,
          "deletions": null,
          "dependencies": null,
          "elementType": [Function],
          "flags": 1,
          "index": 0,
          "key": null,
          "lanes": 0,
          "memoizedProps": {
            "content": [Circular],
          },
          "memoizedState": null,
          "mode": 1,
          "pendingProps": {
            "content": [Circular],
          },
          "ref": null,
          "refCleanup": null,
          "return": [Circular],
          "selfBaseDuration": -0,
          "sibling": null,
          "stateNode": View {
            "_reactInternalInstance": {},
            "_reactInternals": [Circular],
            "context": {},
            "props": {
              "content": [Circular],
            },
            "refs": {},
            "state": null,
            "updater": {
              "enqueueForceUpdate": [Function],
              "enqueueReplaceState": [Function],
              "enqueueSetState": [Function],
              "isMounted": [Function],
            },
          },
          "subtreeFlags": 0,
          "tag": 1,
          "treeBaseDuration": -0,
          "type": [Function],
          "updateQueue": {
            "baseState": null,
            "callbacks": null,
            "firstBaseUpdate": null,
            "lastBaseUpdate": null,
            "shared": {
              "hiddenCallbacks": null,
              "lanes": 0,
              "pending": null,
            },
          },
        },
        "childLanes": 0,
        "deletions": null,
        "dependencies": null,
        "elementType": [Function],
        "flags": 33554433,
        "index": 0,
        "key": null,
        "lanes": 0,
        "memoizedProps": {},
        "memoizedState": null,
        "mode": 1,
        "pendingProps": {},
        "ref": null,
        "refCleanup": null,
        "return": FiberNode {
          "_debugHookTypes": null,
          "_debugInfo": null,
          "_debugNeedsRemount": false,
          "_debugOwner": null,
          "actualDuration": -0,
          "actualStartTime": 1740238588745,
          "alternate": FiberNode {
            "_debugHookTypes": null,
            "_debugInfo": null,
            "_debugNeedsRemount": false,
            "_debugOwner": null,
            "actualDuration": -0,
            "actualStartTime": -1.1,
            "alternate": [Circular],
            "child": null,
            "childLanes": 0,
            "deletions": null,
            "dependencies": null,
            "elementType": null,
            "flags": 0,
            "index": 0,
            "key": null,
            "lanes": 32,
            "memoizedProps": null,
            "memoizedState": {
              "cache": {
                "controller": AbortController {},
                "data": Map {},
                "refCount": 1,
              },
              "element": null,
              "isDehydrated": false,
            },
            "mode": 1,
            "pendingProps": null,
            "ref": null,
            "refCleanup": null,
            "return": null,
            "selfBaseDuration": -0,
            "sibling": null,
            "stateNode": FiberRootNode {
              "_debugRootType": "createRoot()",
              "callbackNode": null,
              "callbackPriority": 0,
              "cancelPendingCommit": null,
              "containerInfo": {
                "children": [
                  {
                    "children": [],
                    "internalInstanceHandle": FiberNode {
                      "_debugHookTypes": null,
                      "_debugInfo": null,
                      "_debugNeedsRemount": false,
                      "_debugOwner": FiberNode {
                        "_debugHookTypes": null,
                        "_debugInfo": null,
                        "_debugNeedsRemount": false,
                        "_debugOwner": [Circular],
                        "actualDuration": -0,
                        "actualStartTime": 1740238588745,
                        "alternate": null,
                        "child": [Circular],
                        "childLanes": 0,
                        "deletions": null,
                        "dependencies": null,
                        "elementType": [Function],
                        "flags": 1,
                        "index": 0,
                        "key": null,
                        "lanes": 0,
                        "memoizedProps": {
                          "content": [Circular],
                        },
                        "memoizedState": null,
                        "mode": 1,
                        "pendingProps": {
                          "content": [Circular],
                        },
                        "ref": null,
                        "refCleanup": null,
                        "return": [Circular],
                        "selfBaseDuration": -0,
                        "sibling": null,
                        "stateNode": View {
                          "_reactInternalInstance": {},
                          "_reactInternals": [Circular],
                          "context": {},
                          "props": {
                            "content": [Circular],
                          },
                          "refs": {},
                          "state": null,
                          "updater": {
                            "enqueueForceUpdate": [Function],
                            "enqueueReplaceState": [Function],
                            "enqueueSetState": [Function],
                            "isMounted": [Function],
                          },
                        },
                        "subtreeFlags": 0,
                        "tag": 1,
                        "treeBaseDuration": -0,
                        "type": [Function],
                        "updateQueue": {
                          "baseState": null,
                          "callbacks": null,
                          "firstBaseUpdate": null,
                          "lastBaseUpdate": null,
                          "shared": {
                            "hiddenCallbacks": null,
                            "lanes": 0,
                            "pending": null,
                          },
                        },
                      },
                      "actualDuration": -0,
                      "actualStartTime": 1740238588726,
                      "alternate": null,
                      "child": null,
                      "childLanes": 0,
                      "deletions": null,
                      "dependencies": null,
                      "elementType": "View",
                      "flags": 0,
                      "index": 0,
                      "key": null,
                      "lanes": 0,
                      "memoizedProps": {
                        "children": undefined,
                        "content": [Circular],
                      },
                      "memoizedState": null,
                      "mode": 1,
                      "pendingProps": {
                        "children": undefined,
                        "content": [Circular],
                      },
                      "ref": null,
                      "refCleanup": null,
                      "return": FiberNode {
                        "_debugHookTypes": null,
                        "_debugInfo": null,
                        "_debugNeedsRemount": false,
                        "_debugOwner": [Circular],
                        "actualDuration": -0,
                        "actualStartTime": 1740238588745,
                        "alternate": null,
                        "child": [Circular],
                        "childLanes": 0,
                        "deletions": null,
                        "dependencies": null,
                        "elementType": [Function],
                        "flags": 1,
                        "index": 0,
                        "key": null,
                        "lanes": 0,
                        "memoizedProps": {
                          "content": [Circular],
                        },
                        "memoizedState": null,
                        "mode": 1,
                        "pendingProps": {
                          "content": [Circular],
                        },
                        "ref": null,
                        "refCleanup": null,
                        "return": [Circular],
                        "selfBaseDuration": -0,
                        "sibling": null,
                        "stateNode": View {
                          "_reactInternalInstance": {},
                          "_reactInternals": [Circular],
                          "context": {},
                          "props": {
                            "content": [Circular],
                          },
                          "refs": {},
                          "state": null,
                          "updater": {
                            "enqueueForceUpdate": [Function],
                            "enqueueReplaceState": [Function],
                            "enqueueSetState": [Function],
                            "isMounted": [Function],
                          },
                        },
                        "subtreeFlags": 0,
                        "tag": 1,
                        "treeBaseDuration": -0,
                        "type": [Function],
                        "updateQueue": {
                          "baseState": null,
                          "callbacks": null,
                          "firstBaseUpdate": null,
                          "lastBaseUpdate": null,
                          "shared": {
                            "hiddenCallbacks": null,
                            "lanes": 0,
                            "pending": null,
                          },
                        },
                      },
                      "selfBaseDuration": -0,
                      "sibling": null,
                      "stateNode": [Circular],
                      "subtreeFlags": 0,
                      "tag": 5,
                      "treeBaseDuration": -0,
                      "type": "View",
                      "updateQueue": null,
                    },
                    "isHidden": false,
                    "props": {
                      "children": undefined,
                      "content": [Circular],
                    },
                    "rootContainerInstance": [Circular],
                    "tag": "INSTANCE",
                    "type": "View",
                  },
                ],
                "createNodeMock": [Function],
                "tag": "CONTAINER",
              },
              "context": {},
              "current": [Circular],
              "effectDuration": 0,
              "entangledLanes": 0,
              "entanglements": [
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
                0,
              ],
              "errorRecoveryDisabledLanes": 0,
              "expirationTimes": [
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
                -1,
              ],
              "expiredLanes": 0,
              "finishedLanes": 0,
              "finishedWork": null,
              "formState": null,
              "hiddenUpdates": [
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
                null,
              ],
              "identifierPrefix": "",
              "incompleteTransitions": Map {},
              "next": null,
              "onCaughtError": [Function],
              "onRecoverableError": [Function],
              "onUncaughtError": [Function],
              "passiveEffectDuration": -0,
              "pendingChildren": null,
              "pendingContext": null,
              "pendingLanes": 0,
              "pingCache": null,
              "pingedLanes": 0,
              "pooledCache": null,
              "pooledCacheLanes": 0,
              "shellSuspendCounter": 0,
              "suspendedLanes": 0,
              "tag": 1,
              "timeoutHandle": -1,
              "warmLanes": 0,
            },
            "subtreeFlags": 0,
            "tag": 3,
            "treeBaseDuration": -0,
            "type": null,
            "updateQueue": {
              "baseState": {
                "cache": {
                  "controller": AbortController {},
                  "data": Map {},
                  "refCount": 1,
                },
                "element": null,
                "isDehydrated": false,
              },
              "callbacks": null,
              "firstBaseUpdate": {
                "callback": null,
                "lane": 32,
                "next": null,
                "payload": {
                  "element": {
                    "$$typeof": Symbol(react.transitional.element),
                    "_owner": null,
                    "_store": {},
                    "key": null,
                    "props": {},
                    "type": [Function],
                  },
                },
                "tag": 0,
              },
              "lastBaseUpdate": {
                "callback": null,
                "lane": 32,
                "next": null,
                "payload": {
                  "element": {
                    "$$typeof": Symbol(react.transitional.element),
                    "_owner": null,
                    "_store": {},
                    "key": null,
                    "props": {},
                    "type": [Function],
                  },
                },
                "tag": 0,
              },
              "shared": {
                "hiddenCallbacks": null,
                "lanes": 0,
                "pending": null,
              },
            },
          },
          "child": [Circular],
          "childLanes": 0,
          "deletions": null,
          "dependencies": null,
          "elementType": null,
          "flags": 1024,
          "index": 0,
          "key": null,
          "lanes": 0,
          "memoizedProps": null,
          "memoizedState": {
            "cache": {
              "controller": AbortController {},
              "data": Map {},
              "refCount": 1,
            },
            "element": {
              "$$typeof": Symbol(react.transitional.element),
              "_owner": null,
              "_store": {},
              "key": null,
              "props": {},
              "type": [Function],
            },
            "isDehydrated": false,
          },
          "mode": 1,
          "pendingProps": null,
          "ref": null,
          "refCleanup": null,
          "return": null,
          "selfBaseDuration": -0,
          "sibling": null,
          "stateNode": FiberRootNode {
            "_debugRootType": "createRoot()",
            "callbackNode": null,
            "callbackPriority": 0,
            "cancelPendingCommit": null,
            "containerInfo": {
              "children": [
                {
                  "children": [],
                  "internalInstanceHandle": FiberNode {
                    "_debugHookTypes": null,
                    "_debugInfo": null,
                    "_debugNeedsRemount": false,
                    "_debugOwner": FiberNode {
                      "_debugHookTypes": null,
                      "_debugInfo": null,
                      "_debugNeedsRemount": false,
                      "_debugOwner": [Circular],
                      "actualDuration": -0,
                      "actualStartTime": 1740238588745,
                      "alternate": null,
                      "child": [Circular],
                      "childLanes": 0,
                      "deletions": null,
                      "dependencies": null,
                      "elementType": [Function],
                      "flags": 1,
                      "index": 0,
                      "key": null,
                      "lanes": 0,
                      "memoizedProps": {
                        "content": [Circular],
                      },
                      "memoizedState": null,
                      "mode": 1,
                      "pendingProps": {
                        "content": [Circular],
                      },
                      "ref": null,
                      "refCleanup": null,
                      "return": [Circular],
                      "selfBaseDuration": -0,
                      "sibling": null,
                      "stateNode": View {
                        "_reactInternalInstance": {},
                        "_reactInternals": [Circular],
                        "context": {},
                        "props": {
                          "content": [Circular],
                        },
                        "refs": {},
                        "state": null,
                        "updater": {
                          "enqueueForceUpdate": [Function],
                          "enqueueReplaceState": [Function],
                          "enqueueSetState": [Function],
                          "isMounted": [Function],
                        },
                      },
                      "subtreeFlags": 0,
                      "tag": 1,
                      "treeBaseDuration": -0,
                      "type": [Function],
                      "updateQueue": {
                        "baseState": null,
                        "callbacks": null,
                        "firstBaseUpdate": null,
                        "lastBaseUpdate": null,
                        "shared": {
                          "hiddenCallbacks": null,
                          "lanes": 0,
                          "pending": null,
                        },
                      },
                    },
                    "actualDuration": -0,
                    "actualStartTime": 1740238588726,
                    "alternate": null,
                    "child": null,
                    "childLanes": 0,
                    "deletions": null,
                    "dependencies": null,
                    "elementType": "View",
                    "flags": 0,
                    "index": 0,
                    "key": null,
                    "lanes": 0,
                    "memoizedProps": {
                      "children": undefined,
                      "content": [Circular],
                    },
                    "memoizedState": null,
                    "mode": 1,
                    "pendingProps": {
                      "children": undefined,
                      "content": [Circular],
                    },
                    "ref": null,
                    "refCleanup": null,
                    "return": FiberNode {
                      "_debugHookTypes": null,
                      "_debugInfo": null,
                      "_debugNeedsRemount": false,
                      "_debugOwner": [Circular],
                      "actualDuration": -0,
                      "actualStartTime": 1740238588745,
                      "alternate": null,
                      "child": [Circular],
                      "childLanes": 0,
                      "deletions": null,
                      "dependencies": null,
                      "elementType": [Function],
                      "flags": 1,
                      "index": 0,
                      "key": null,
                      "lanes": 0,
                      "memoizedProps": {
                        "content": [Circular],
                      },
                      "memoizedState": null,
                      "mode": 1,
                      "pendingProps": {
                        "content": [Circular],
                      },
                      "ref": null,
                      "refCleanup": null,
                      "return": [Circular],
                      "selfBaseDuration": -0,
                      "sibling": null,
                      "stateNode": View {
                        "_reactInternalInstance": {},
                        "_reactInternals": [Circular],
                        "context": {},
                        "props": {
                          "content": [Circular],
                        },
                        "refs": {},
                        "state": null,
                        "updater": {
                          "enqueueForceUpdate": [Function],
                          "enqueueReplaceState": [Function],
                          "enqueueSetState": [Function],
                          "isMounted": [Function],
                        },
                      },
                      "subtreeFlags": 0,
                      "tag": 1,
                      "treeBaseDuration": -0,
                      "type": [Function],
                      "updateQueue": {
                        "baseState": null,
                        "callbacks": null,
                        "firstBaseUpdate": null,
                        "lastBaseUpdate": null,
                        "shared": {
                          "hiddenCallbacks": null,
                          "lanes": 0,
                          "pending": null,
                        },
                      },
                    },
                    "selfBaseDuration": -0,
                    "sibling": null,
                    "stateNode": [Circular],
                    "subtreeFlags": 0,
                    "tag": 5,
                    "treeBaseDuration": -0,
                    "type": "View",
                    "updateQueue": null,
                  },
                  "isHidden": false,
                  "props": {
                    "children": undefined,
                    "content": [Circular],
                  },
                  "rootContainerInstance": [Circular],
                  "tag": "INSTANCE",
                  "type": "View",
                },
              ],
              "createNodeMock": [Function],
              "tag": "CONTAINER",
            },
            "context": {},
            "current": [Circular],
            "effectDuration": 0,
            "entangledLanes": 0,
            "entanglements": [
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
              0,
            ],
            "errorRecoveryDisabledLanes": 0,
            "expirationTimes": [
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
              -1,
            ],
            "expiredLanes": 0,
            "finishedLanes": 0,
            "finishedWork": null,
            "formState": null,
            "hiddenUpdates": [
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
              null,
            ],
            "identifierPrefix": "",
            "incompleteTransitions": Map {},
            "next": null,
            "onCaughtError": [Function],
            "onRecoverableError": [Function],
            "onUncaughtError": [Function],
            "passiveEffectDuration": -0,
            "pendingChildren": null,
            "pendingContext": null,
            "pendingLanes": 0,
            "pingCache": null,
            "pingedLanes": 0,
            "pooledCache": null,
            "pooledCacheLanes": 0,
            "shellSuspendCounter": 0,
            "suspendedLanes": 0,
            "tag": 1,
            "timeoutHandle": -1,
            "warmLanes": 0,
          },
          "subtreeFlags": 33554435,
          "tag": 3,
          "treeBaseDuration": -0,
          "type": null,
          "updateQueue": {
            "baseState": {
              "cache": {
                "controller": AbortController {},
                "data": Map {},
                "refCount": 1,
              },
              "element": {
                "$$typeof": Symbol(react.transitional.element),
                "_owner": null,
                "_store": {},
                "key": null,
                "props": {},
                "type": [Function],
              },
              "isDehydrated": false,
            },
            "callbacks": null,
            "firstBaseUpdate": null,
            "lastBaseUpdate": null,
            "shared": {
              "hiddenCallbacks": null,
              "lanes": 0,
              "pending": null,
            },
          },
        },
        "selfBaseDuration": -0,
        "sibling": null,
        "stateNode": null,
        "subtreeFlags": 1,
        "tag": 0,
        "treeBaseDuration": -0,
        "type": [Function],
        "updateQueue": null,
      },
      "_store": {},
      "key": null,
      "props": {},
      "type": [Function],
    }
  }
/>
`;

exports[`components rendering should render \`OverKeyboardView\` 1`] = `
<OverKeyboardView
  visible={true}
>
  <View
    style={
      {
        "backgroundColor": "black",
        "height": 20,
        "width": 20,
      }
    }
  />
</OverKeyboardView>
`;
